arangod/Replication2/ReplicatedLog/LogLeader.cpp

```diff
@@ -273,69 +273,80 @@ void replicated_log::LogLeader::executeAppendEntriesRequests(
-                                                 logMetrics,
-                                                 self->_scheduler.get());
-                  } else {
-                    if (follower == nullptr) {
-                      LOG_TOPIC("6f490", DEBUG, Logger::REPLICATION2)
-                          << "follower already gone.";
-                    } else {
-                      LOG_CTX("de300", DEBUG, follower->logContext)
-                          << "parent log already gone, messageId = "
-                          << messageId;
-                    }
+            .thenFinal([weakParentLog = req->_parentLog,
+                        followerWeak = req->_follower, lastIndex = lastIndex,
+                        currentCommitIndex, currentLITK,
+                        currentTerm = logLeader->_currentTerm, messageId,
+                        startTime, logMetrics](
+                           futures::Try<AppendEntriesResult>&& res) noexcept {
+              // This has to remain noexcept, because the code below is not
+              // exception safe
+              auto const endTime = std::chrono::steady_clock::now();
+
+              auto self = weakParentLog.lock();
+              auto follower = followerWeak.lock();
+              if (self != nullptr && follower != nullptr) {
+                using namespace std::chrono_literals;
+                auto const duration = endTime - startTime;
+                self->_logMetrics->replicatedLogAppendEntriesRttUs->count(
+                    duration / 1us);
+                LOG_CTX("8ff44", TRACE, follower->logContext)
+                    << "received append entries response, messageId = "
+                    << messageId;
+                auto responseResult = basics::catchToResultT(
+                    [&]() -> std::pair<std::vector<std::optional<
+                                           PreparedAppendEntryRequest>>,
+                                       ResolvedPromiseSet> {
+                      auto guarded = self->acquireMutex();
+                      if (!guarded->_didResign) {
+                        return guarded->handleAppendEntriesResponse(
+                            *follower, lastIndex, currentCommitIndex,
+                            currentLITK, currentTerm, std::move(res),
+                            endTime - startTime, messageId);
+                      } else {
+                        LOG_CTX("da116", DEBUG, follower->logContext)
+                            << "received response from follower but leader "
+                               "already resigned, messageId = "
+                            << messageId;
+                      }
+                      return {};
+                    });
+                if (responseResult.fail()) {
+                  LOG_TOPIC("a32de", DEBUG, Logger::REPLICATION2)
+                      << "appendEntries failed, follower already gone: "
+                      << responseResult.result();
```

Comment:
The log message might not be accurate, unless you move it into the `if` below

Reply:
Good catch, addressed in https://github.com/arangodb/arangodb/pull/20628/commits/7f9c83b10f065a508acf68dc96cb815bd452dcd0